lang/src/main/kotlin/org/partiql/lang/eval/EvaluatingCompiler.kt

```diff
@@ -1599,25 +1585,17 @@ internal class EvaluatingCompiler(
             )
         }
 
-        // `Number.toLong()` (used below) does *not* cause an overflow exception if the underlying [Number]
-        // implementation (i.e. Decimal or BigInteger) exceeds the range that can be represented by Longs.
-        // This can cause very confusing behavior if the user specifies a LIMIT value that exceeds
-        // Long.MAX_VALUE, because no results will be returned from their query.  That no overflow exception
-        // is thrown is not a problem as long as PartiQL's restriction of integer values to +/- 2^63 remains.
-        // We throw an exception here if the value exceeds the supported range (say if we change that
-        // restriction or if a custom [ExprValue] is provided which exceeds that value).
-        val limitIonValue = limitExprValue.ionValue as IonInt
-        if (limitIonValue.integerSize == IntegerSize.BIG_INTEGER) {
+        val originalLimitValue = limitExprValue.numberValue()
+        val limitValue = originalLimitValue.toLong()
+        if (originalLimitValue != limitValue as Number) { // Make sure `Number.toLong()` is a lossless transformation
             err(
-                "IntegerSize.BIG_INTEGER not supported for LIMIT values",
+                "Too large integer provided as LIMIT value",
```

Comment:
Can we say something more like `LIMIT value exceeds Long.MAX_VALUE` or something? I'm not sure what "too large" means without the context of the previous comment that was here.